src/core/UserModule.h

```diff
@@ -28,11 +28,11 @@ class StaticModuleNameStack
 class UserModule : public AbstractModule, public ASTNode
 {
 public:
-  UserModule(const char *name, const Location& loc) : ASTNode(loc), name(name) { }
-  UserModule(const char *name, const Feature& feature, const Location& loc) : AbstractModule(feature), ASTNode(loc), name(name) { }
+  UserModule(const char *name, const Location& loc, AssignmentList& parameters) :
+    ASTNode(loc), name(name), parameters(parameters) { }
 
   std::shared_ptr<AbstractNode> instantiate(const std::shared_ptr<const Context>& defining_context, const ModuleInstantiation *inst, const std::shared_ptr<const Context>& context) const override;
-  void print(std::ostream& stream, const std::string& indent) const override;
+  virtual void print(std::ostream& stream, const std::string& indent) const override;
```

Comment:
if it was `override`, why we need `virtual` here?

Reply:
See above comment about my C++ being weak.